src/PyDynamic/uncertainty/propagate_convolution.py

```diff
@@ -150,3 +162,18 @@ def convolve_unc(x1, U1, x2, U2, mode="full"):
         raise ValueError(f'convolve_unc: Mode "{mode}" is not supported.')
 
     return conv, Uconv
+
+
+def _ensure_cov_matrix(unc_array):
+    """
+    Converts 1D-arrays of standard uncertainties into the corresponding
+    (diagonal) covariance matrix by *square*+diag.
+
+    Does not modify inputs which are 2D-arrays or None.
+    """
+
+    # squeeze() ensures proper execution on arrays with only one dimension of non-zero length
+    if unc_array is not None and len(np.squeeze(unc_array).shape) == 1:
+        unc_array = np.diag(np.square(np.squeeze(unc_array)))
```

Comment:
If we really want to support lists and tuples, we should change the docstring and test these cases as well.

Reply:
Resolved by our discussion, where we agreed on refactoring `is_vector()`.